contingency/contingency-api/src/main/java/com/powsybl/contingency/json/AbstractEquipmentCriterionContingencyListDeserializer.java

```diff
@@ -69,7 +70,9 @@ protected boolean deserializeCommonAttributes(JsonParser parser, Deserialization
                 return true;
             }
             case "type" -> {
-                if (!parser.nextTextValue().equals(expectedType)) {
+                // parser.nextTextValue() could returns null
+                String typeStr = Objects.requireNonNull(parser.nextTextValue());
```

Comment:
Thank you for your contribution!

Here, calling `Objects.requireNonNull(...)` will also throw a NPE if the type is null (but the cause will indeed be different). 
It may be better to reverse the test (`!expectedType.equals(typeStr)`) because `expectedType` is not an input. In this case, an `IllegalStateException` with a clear message will be thrown.